scripts/preprocess_wikitables.py

```diff
@@ -0,0 +1,52 @@
+import json
+import logging
+import os
+import sys
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir))))
+logging.basicConfig(format='%(asctime)s - %(levelname)s - %(name)s - %(message)s',
+                    level=logging.INFO)
+from allennlp.commands.train import datasets_from_params
+from allennlp.common import Params
+from allennlp.data import Instance
+
+
+def main(params: Params, outdir: str):
+    os.makedirs(outdir, exist_ok=True)
+    all_datasets = datasets_from_params(params)
+    for name, dataset in all_datasets.items():
+        with open(outdir + name + '.jsonl', 'w') as outfile:
+            for instance in iter(dataset):
+                outfile.write(to_json_line(instance) + '\n')
+
+
+def to_json_line(instance: Instance):
+    json_obj = {}
+    question_tokens = instance.fields['question'].tokens
+    json_obj['question_tokens'] = [{'text': token.text, 'lemma': token.lemma_}
+                                   for token in question_tokens]
+    json_obj['table_lines'] = instance.fields['table_metadata'].metadata
+
+    action_map = {i: action.rule for i, action in enumerate(instance.fields['actions'].field_list)}
+    targets = []
+    for target_sequence in instance.fields['target_action_sequences'].field_list:
+        targets.append([])
+        for target_index_field in target_sequence.field_list:
+            targets[-1].append(action_map[target_index_field.sequence_index])
+
+    json_obj['target_action_sequences'] = targets
+
+    entity_texts = []
+    for entity_text in instance.fields['table'].entity_texts:
+        tokens = [{'text': token.text, 'lemma': token.lemma_} for token in entity_text]
+        entity_texts.append(tokens)
+    json_obj['entity_texts'] = entity_texts
+    json_obj['linking_features'] = instance.fields['table'].linking_features
```

Comment:
Looks like these will be huge sparse arrays. May be you can represent them as dicts instead?

Reply:
They are 7 numbers per token, per entity.  It's not a huge deal to store them this way, and the logic to go to and from sparse dicts isn't really worth it, I think.